File: models/Appointment.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const appointmentSchema = new Schema({
    doctorUsername: {
        type: String,
        required: true
    },
    patientUsername: {
        type: String,
        required: false
    },
    date: {
        type: String,
        required: true
    },
    startHour: {
        type: Number,
        required: true
    },
    endHour: {
        type: Number,
        required: true
    },
    isFollowUp: {
        type: Boolean,
        required: false
    },
    status: {
        type: String,
        enum: ['unreserved', 'upcoming', 'completed', 'cancelled', 'rescheduled', 'follow-up Requested', 'followUpAccepted', 'followUpRevoked'],
        default: 'unreserved',
        required: true
    }
}, {timestamps: true});

const Appointment = mongoose.model('Appointment', appointmentSchema, 'Appointments');
module.exports = Appointment;

File: Clinic Fron End/lib/Models/Appointment.dart
class Appointment {
  String id = "";
  String doctor = "";
  String patient = "";
  String date = "";
  double startHour = 0;
  double endHour = 0;
  String status = "";

  Appointment.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    doctor = json['doctor'];
    patient = json['patient'];
    date = json['date'];
    startHour = json['startHour'];
    endHour = json['endHour'];
    status = json['status'];
  }
}
